mapper/orm/src/main/java/org/hibernate/search/mapper/orm/work/impl/SearchWorkspaceImpl.java

```diff
@@ -23,12 +24,12 @@ public SearchWorkspaceImpl(PojoScopeWorkspace delegate) {
 
 	@Override
 	public void mergeSegments() {
-		Futures.unwrappedExceptionJoin( mergeSegmentsAsync() );
+		Futures.unwrappedExceptionJoin( delegate.mergeSegments( OperationSubmitter.BLOCKING ) );
 	}
 
 	@Override
 	public CompletableFuture<?> mergeSegmentsAsync() {
-		return delegate.mergeSegments();
+		return delegate.mergeSegments( OperationSubmitter.REJECTED_EXECUTION_EXCEPTION );
```

Comment:
:+1: 

Could you add something to the "behavior" section of the migration guide, though? This is technically a breaking change. 

No need to list every single method, but something along the lines of "`*Async` methods will no longer block if internal queues of operations are full, but will throw `RejectedOperationException` instead."